qiskit/quantum_info/states/stabilizerstate.py

```diff
@@ -0,0 +1,587 @@
+            np.array: The Numpy vector array of probabilities.
+        """
+        probs_dict = self.probabilities_dict(qargs, decimals)
+        if qargs is None:
+            qargs = range(self.clifford.num_qubits)
+        probs = np.zeros(2 ** len(qargs))
+
+        for key, value in probs_dict.items():
+            place = int(key, 2)
+            probs[place] = value
+
+        return probs
+
+    def probabilities_dict(self, qargs=None, decimals=None):
+        """Return the subsystem measurement probability dictionary.
+
+        Measurement probabilities are with respect to measurement in the
+        computation (diagonal) basis.
+
+        This dictionary representation uses a Ket-like notation where the
+        dictionary keys are qudit strings for the subsystem basis vectors.
+        If any subsystem has a dimension greater than 10 comma delimiters are
+        inserted between integers so that subsystems can be distinguished.
+
+        Args:
+            qargs (None or list): subsystems to return probabilities for,
+                if None return for all subsystems (Default: None).
+            decimals (None or int): the number of decimal places to round
+                values. If None no rounding is done (Default: None).
+
+        Returns:
+            dict: The measurement probabilities in dict (ket) form.
+        """
+        if qargs is None:
+            qubits = range(self.clifford.num_qubits)
+        else:
+            qubits = qargs
+
+        outcome = ["X"] * len(qubits)
+        outcome_prob = 1.0
+        probs = {}  # probabilities dictionary
+
+        self._get_probablities(qubits, outcome, outcome_prob, probs)
+
+        if decimals is not None:
+            for key, value in probs.items():
+                probs[key] = round(value, decimals)
+
+        return probs
+
+    def reset(self, qargs=None):
+        """Reset state or subsystems to the 0-state.
+
+        Args:
+            qargs (list or None): subsystems to reset, if None all
+                                  subsystems will be reset to their 0-state
+                                  (Default: None).
+
+        Returns:
+            StabilizerState: the reset state.
+
+        Additional Information:
+            If all subsystems are reset this will return the ground state
+            on all subsystems. If only some subsystems are reset this
+            function will perform a measurement on those subsystems and
+            evolve the subsystems so that the collapsed post-measurement
+            states are rotated to the 0-state. The RNG seed for this
+            sampling can be set using the :meth:`seed` method.
+        """
+        # Resetting all qubits does not require sampling or RNG
+        if qargs is None:
+            return StabilizerState(Clifford((np.eye(2 * self.clifford.num_qubits))))
+
+        randbits = self._rng.integers(2, size=len(qargs))
+        ret = self.copy()
+        bit = 0
+        for qubit in qargs:
```

Comment:
Rather than manually increment the counter you can do
```suggestion
        for bit, qubit in enumerate(qargs):
```